src/api/segm/ISemanticSegmentation.cpp

```diff
@@ -0,0 +1,44 @@
+/**
+ * @copyright Copyright (c) 2025 B-com http://www.b-com.com/
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "api/segm/ISemanticSegmentation.h"
+#include "core/Log.h"
+
+namespace SolAR {
+using namespace datastructure;
+namespace api {
+namespace segm {
+
+FrameworkReturnCode ISemanticSegmentation::segment(const std::vector<SRef<Image>>& images, std::vector<SRef<Image>>& masks)
+{
+    masks.resize(images.size());
+    bool hasError = false;
+    for (size_t i = 0; i < images.size(); ++i) {
+        if (segment(images[i], masks[i]) != FrameworkReturnCode::_SUCCESS) {
+            hasError = true;
+            break;
```

Comment:
Since you exit on first error, you could avoid this variable ?
```cpp
masks.resize(images.size());
for (size_t i = 0; i < images.size(); ++i) {
    if (segment(images[i], masks[i]) != FrameworkReturnCode::_SUCCESS) {
        LOG_ERROR("ISemanticSegmentation::segment - semantic segmentation encountered errors.");
        return FrameworkReturnCode::_ERROR_;
    }
}
    return FrameworkReturnCode::_SUCCESS;
```

Reply:
Also, I don't know here, what do you think about resetting the `masks` vector to reuturn an empty result?